### PR TITLE
[MM-15067] Permit Bot accounts to create user access tokens if Enable Personal Access Tokens set to "false"

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR addresses the case when System Console > INTEGRATIONS > Enable Personal Access Tokens is set to "false".  In this situation, we should only restrict the generation of access tokens for non-Bot accounts.

The solution proposed is to check that the user is not a Bot , regardless of the "Enable Personal Access Tokens" setting.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15067

### CODE REVIEW

File: app/session.go [hunk 1]
```diff
@@ -242,15 +242,17 @@ func (a *App) UpdateLastActivityAtIfNeeded(session model.Session) {
...
+	user, err := a.Srv.Store.User().Get(token.UserId)
+
+	if !*a.Config().ServiceSettings.EnableUserAccessTokens && !user.IsBot {
 		return nil, model.NewAppError("CreateUserAccessToken", "app.user_access_token.disabled", nil, "", http.StatusNotImplemented)
 	}
 
 	token.Token = model.NewId()
 
 	uchan := make(chan store.StoreResult, 1)
 	go func() {
```
Comment (crspeller) [inline app/session.go]: If we need to block now we should remove this unnecessary goroutine/channel.

Comment (jfrerich) [reply app/session.go]: I'll remove both the go routine and channel usage.

File: app/session.go [hunk 2]
```diff
@@ -242,29 +241,24 @@ func (a *App) UpdateLastActivityAtIfNeeded(session model.Session) {
 }
 
 func (a *App) CreateUserAccessToken(token *model.UserAccessToken) (*model.UserAccessToken, *model.AppError) {
-	if !*a.Config().ServiceSettings.EnableUserAccessTokens {
+
+	user, err := a.Srv.Store.User().Get(token.UserId)
+
+	if !*a.Config().ServiceSettings.EnableUserAccessTokens && !user.IsBot {
```
Comment (levb) [inline app/session.go]: Couldn't `user` be a nil here, since there is no immediate error check?

Comment (jfrerich) [reply app/session.go]: Should the err value in this call cover the nil user case?  Looking at the previous implementation, it looks like user == nil was not checked.   I would like to locate the Get Function call, in the code base, but haven't found it yet.  Where is this method defined?

The err check is later in the function on rows 259, 260, but I think it should be moved under the call to create the user (line 245).

Comment (jasonblais) [reply app/session.go]: @levb Thoughts on the above note from @jfrerich?

**Comment by hanzei** [comment]
Hey @levb,

Gently reminder about the above discussion.

File: app/session.go (hunk 2 above)
Comment (hanzei) [reply app/session.go]: @jfrerich What are you thoughts on this?

Comment (jfrerich) [reply app/session.go]: @hanzei I agree with Lev and will get the change added and resubmit the PR.

**Comment by amyblais** [comment]
@levb was this cherry-picked to v5.12?

**Comment by levb** [comment]
Done now @amyblais

**Pull request merged**